src/client-pool.ts

```diff
@@ -1,46 +1,118 @@
 	public static create<T>(
 		alias: string,
 		size: number,
+		config: ClientConfig,
 		createClient: () => T,
 	): T {
 		let connectionPool: ConnectionPool | undefined =
 			ClientPool.clientsPools.get(alias);
 		if (!connectionPool) {
+			const client = createClient();
+			(client as any).poolPosition = -1;
 			connectionPool = {
 				position: -1,
-				connections: [createClient()],
+				config,
+				factory: createClient,
+				connections: [client],
 			};
 			ClientPool.clientsPools.set(alias, connectionPool);
 
-			connectionPool.proxy = new Proxy(connectionPool.connections[0], {
-				get(_target, name) {
-					connectionPool!.position =
-						(connectionPool!.position + 1) % connectionPool!.connections.length;
-					const client = connectionPool!.connections[connectionPool!.position];
-					if (name in client) {
-						if (typeof client[name as keyof Con] === 'function') {
-							return client[name as keyof Con].bind(client);
-						}
-						return client[name as keyof Con];
-					}
-				},
-			});
+			connectionPool.proxy = factoryPoolProxy(connectionPool);
 		}
 
 		while (connectionPool.connections.length < size) {
-			connectionPool.connections.push(createClient());
+			this.addConnection(alias, createClient);
 		}
 
 		return connectionPool.proxy;
 	}
+	public static remove(alias: string, position: number) {
+		const pool = ClientPool.clientsPools.get(alias);
+
+		if (!pool) {
+			throw new Error(`Not found pool ${alias} to remove`);
+		}
+
+		let connections: any[] | undefined = pool?.connections;
+
+		if (!connections && !connections[position]) {
+			throw new Error(`Not found connections from ${alias} to remove`);
+		}
+		connections[position].client.close();
+		closeClient(connections[position].client);
+
+		connections = connections.filter((_value, index) => {
+			return index !== position;
+		});
+
+		pool.connections = connections;
+		ClientPool.clientsPools.set(alias, pool);
+	}
+	public static addConnection<T>(alias: string, createClient: () => T): T {
+		const pool = ClientPool.clientsPools.get(alias);
+		if (!pool) {
+			throw new Error(`Not found pool ${alias} to add a new connection`);
+		}
+		const client = createClient() as any;
+
+		pool.connections.push(client);
+		return pool.proxy;
+	}
+
+	public static updateConnection<T>(
+		alias: string,
+		position: number,
+		createClient: () => T,
+	): T {
+		const pool = ClientPool.clientsPools.get(alias);
+		if (!pool) {
+			throw new Error(`Not found pool ${alias} to add a new connection`);
+		}
+
+		const client = createClient as any;
+		pool.connections[position] = client;
+		return pool.proxy;
+	}
+
+	public static renewConnect<T>(alias: string, client: Client): T {
+		const position = (client as any).poolPosition;
+		client.close();
+		const pool = ClientPool.clientsPools.get(alias);
+		if (!pool) {
+			throw new Error(`Not found pool ${alias} to renew connection`);
+		}
+		return this.updateConnection(alias, position, pool.factory());
```

Comment:
Antes desse If, seria legal checar se a posição atual do pool é igual à conexão, se for, chamar updateConnection, se não for, só retornar a posição atual do pool. Isso porque em um cenário de alta concorrência, pode ser que seja solicitada a renovação de uma mesma posição do pool devido a diferentes rewuests que falharam de uma mesma conexão, mas só o primeiro precisa ser atendido